### ISSUE TITLE
jsdom execution support

### ISSUE THREAD

**Issue by dmatteo** [systemjs/systemjs#821]
I'm running SystemJS in node, as part of an effort of migrating all our codebase from RequireJS to SystemJS

I have a working jsdom instance created with 
```javascript
    jsdomify = require('jsdomify');
    jsdomify.create();
```

And I'm including System and configuring accordingly
```javascript
    System = require('systemjs');
    System.config({baseURL: 'file://' + process.cwd() + "/app/assets/javascripts"});
    require('../../../config.js');
```

However I get a nasty error when I try to import an AMD module (CommonJS works fine)
```javascript
  it('should load jquery', () => {
    return System.import('jquery').then((module) => {
      return expect(module, 'to be defined');
    })
  });
```

And this is the aforementioned error:
```
AMD module file:///Users/domo/projects/podio/frontend-rails/app/assets/javascripts/jspm_packages/github/components/jquery@2.1.3.js did not define
Error loading file:///Users/domo/projects/podio/frontend-rails/app/assets/javascripts/jspm_packages/github/components/jquery@2.1.3.js
```

FYI, `jquery@2.1.3.js' goes like this (any other AMD module I tried has the same behaviour though):
```javascript
define(["github:components/jquery@2.1.3/jquery"], function(main) {
  return main;
});
```

Looking forward to understand what's going on.
Cheers!

**Comment by guybedford** [comment]
@dmatteo thanks for posting. Any help replicating would be appreciated. To dig into it... are you running these tests in Node? With jsdom? Note that jQuery is treated as a global module by default on jspm I believe due to https://github.com/jspm/registry/blob/master/package-overrides/github/components/jquery%402.0.3.json. Are you somehow overriding the format with meta configuration here?

**Comment by Munter** [comment]
No overrides. The jquery 2.1.3 install is plain as can be

**Comment by Munter** [comment]
Example repository: https://github.com/Munter/issue-systemjs-821

**Comment by dmatteo** [comment]
@guybedford 

My setup:
```
SystemJS v0.19.3
node v0.10.26
mocha v2.2.4
jsdomify v0.6.1
```

**Comment by guybedford** [comment]
Thanks for reporting this one. I've renamed this issue to reflect what I suspect is the issue.

This is likely due to the execution method defined in https://github.com/systemjs/systemjs/blob/master/lib/global-eval.js#L74 not working in jsdom.

**Comment by dmatteo** [comment]
Cool, I'll prepare a PR for that.

Thanks

**Comment by guybedford** [comment]
:+1:

Note that the issue with `new Function` execution is that free globals are not assigned to the `window`, which is why we needed that. These cases may still crop up in global executions, but may not be an issue either.

**Comment by dmatteo** [comment]
The idea was to check if I'm in a node environment with 
`typeof exports === 'object' && typeof exports.nodeName !== 'string'`

Now, that works, but I'm not sure why this [this test](https://github.com/systemjs/systemjs/blob/master/test/test.js#L62-L67) fails, being a simple `var foo = 'bar';` 

```
Module: SystemJS Test: Global script with var syntax
Wrong global value
    at /Users/domo/projects/systemjs/test/test.js:50:5
    at tryCatchReject (/Users/domo/projects/systemjs/node_modules/when/es6-shim/Promise.js:1183:30)
    at runContinuation1 (/Users/domo/projects/systemjs/node_modules/when/es6-shim/Promise.js:1142:4)
    at Fulfilled.when (/Users/domo/projects/systemjs/node_modules/when/es6-shim/Promise.js:930:4)
    at Pending.run (/Users/domo/projects/systemjs/node_modules/when/es6-shim/Promise.js:821:13)
    at Scheduler._drain (/Users/domo/projects/systemjs/node_modules/when/es6-shim/Promise.js:97:19)
    at Scheduler.drain (/Users/domo/projects/systemjs/node_modules/when/es6-shim/Promise.js:62:9)
    at process._tickCallback (node.js:415:13)
```

Any idea?

**Comment by guybedford** [comment]
@dmatteo exactly because of https://github.com/systemjs/systemjs/issues/821#issuecomment-145847184.

I'd suggest ensuring the environment detection only applies to jsdom and not Node in general?

**Comment by dmatteo** [comment]
Right... I'll think of something more specific

**Comment by dmatteo** [comment]
I'm trying with this now
```
isBrowser && typeof process != 'undefined' && typeof process.versions != 'undefined' && typeof process.versions.node != 'undefined'
```

Which is fairly specific to a node browser instance (e.g. jsdom).
Still that test is failing.

It shouldn't be affected at all, or am I missing something?

**Comment by dmatteo** [comment]
Uhm, apparently I have to run `npm run build` before `npm run test` for the changes to get applied.

Is that intended?

**Comment by guybedford** [comment]
This should now be working from https://github.com/systemjs/systemjs/commit/808c5cddd9625f56648e4713cc05598e03ab53dd. Let me know if you have any further issues here at all.

**Comment by dmatteo** [comment]
Ok, thanks

**Comment by guybedford** [comment]
Released in 0.19.4. Let me know how this works out for you.
